controllers/component_controller.go

```diff
@@ -47,10 +60,197 @@ type ComponentReconciler struct {
+			}
+			if hasApplication.Status.Devfile != "" {
+				// Get the devfile of the hasApp CR
+				hasAppDevfileData, err := devfile.ParseDevfileModel(hasApplication.Status.Devfile)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to parse the devfile from Application, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				err = r.updateApplicationDevfileModel(hasAppDevfileData, hasComponent)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to update the HAS Application Devfile model %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, nil
+				}
+
+				yamlHASCompData, err := yaml.Marshal(hasCompDevfileData)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to marshall the Component devfile, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				hasComponent.Status.Devfile = string(yamlHASCompData)
+
+				// Update the HASApp CR with the new devfile
+				yamlHASAppData, err := yaml.Marshal(hasAppDevfileData)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to marshall the Application devfile, exiting reconcile loop %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+				hasApplication.Status.Devfile = string(yamlHASAppData)
+				err = r.Status().Update(ctx, &hasApplication)
+				if err != nil {
+					log.Error(err, "Unable to update Application")
+					// if we're unable to update the Application CR, then  we need to err out
+					// since we need to save a reference of the Component in Application
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+
+				r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, nil)
+
+				log.Info(fmt.Sprintf("Updating the labels for HAS Component %v", req.NamespacedName))
+				hasComponentLabels := make(map[string]string)
+				hasComponentLabels["appstudio.has/application"] = hasComponent.Spec.Application
+				hasComponentLabels["appstudio.has/component"] = hasComponent.Spec.ComponentName
+				hasComponent.SetLabels(hasComponentLabels)
+				err = r.Client.Update(ctx, &hasComponent)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("Unable to update Component with the required labels %v", req.NamespacedName))
+					r.SetCreateConditionAndUpdateCR(ctx, &hasComponent, err)
+					return ctrl.Result{}, err
+				}
+			} else {
+				log.Error(err, fmt.Sprintf("Application devfile model is empty. Before creating a Component, an instance of Application should be created, exiting reconcile loop %v", req.NamespacedName))
+				err := fmt.Errorf("application devfile model is empty. Before creating a Component, an instance of Application should be created")
```

Comment:
Thinking out loud here: Should we surface some of these error messages in the `kubectl describe ...` output and also add a `Status` field to the `kubectl get hascomp` / `kubectl get hasapp` output?

What do you think? We don't need to do this in your PR, but should probably open an issue if we think it's worthwhile.

Reply:
so would it be empty most of the time if there is no err?

Reply:
My thinking was it would be set to `Created` / `Updated` / `Error`, etc? But I dunno if that'd be useful.

Reply:
created an [issue](https://issues.redhat.com/browse/DEVHAS-49)

Reply:
Sounds good, thanks. Let's discuss on today's call.